scan.go

```diff
@@ -221,12 +242,29 @@ func (s *ScanService) Do() (*ScanCursor, error) {
 		path += "/" + strings.Join(typesPart, ",")
 	}
 
+	// Define the body of the request
+	var body map[string]interface{}
+
+	if s.source != nil {
+		body = s.source
+	} else {
+		src, err := s.searchSource.Source()
+
+		if err != nil {
+			return nil, err
+		}
+
+		// searchSource.Source() actually returns a map, yet its return type
+		// is specified as interface...
+		body = src.(map[string]interface{})
+	}
+
 	// Search
 	path += "/_search"
 
 	// Parameters
 	params := make(url.Values)
-	if !s.searchSource.hasSort() {
+	if _, ok := body["sort"]; !ok {
```

Comment:
Yes, I think we're mostly saying the same thing: If I'm using Source, there should be no forced internal defeat of the data I intended to set in source, and I also shouldn't expect that the DSL will modify anything I set via Source.

I think the change you suggested makes the most sense, just remove the hasSort check.